metrics/sinks/stackdriver/metadata.go

```diff
@@ -0,0 +1,225 @@
+		ValueType:  "DOUBLE",
+		Name:       "kubernetes.io/container/cpu/limit_cores",
+	}
+
+	memoryContainerUsedBytesMD = &metricMetadata{
+		MetricKind: "GAUGE",
+		ValueType:  "INT64",
+		Name:       "kubernetes.io/container/memory/used_bytes",
+	}
+
+	memoryRequestedBytesMD = &metricMetadata{
+		MetricKind: "GAUGE",
+		ValueType:  "INT64",
+		Name:       "kubernetes.io/container/memory/requested_bytes",
+	}
+
+	memoryLimitBytesMD = &metricMetadata{
+		MetricKind: "GAUGE",
+		ValueType:  "INT64",
+		Name:       "kubernetes.io/container/memory/limit_bytes",
+	}
+
+	restartCountMD = &metricMetadata{
+		MetricKind: "CUMULATIVE",
+		ValueType:  "INT64",
+		Name:       "kubernetes.io/container/restart_count",
+	}
+
+	// Pod metrics
+
+	volumeUsedBytesMD = &metricMetadata{
+		MetricKind: "GAUGE",
+		ValueType:  "INT64",
+		Name:       "kubernetes.io/pod/volume/used_bytes",
+	}
+
+	volumeRequestedBytesMD = &metricMetadata{
+		MetricKind: "GAUGE",
+		ValueType:  "INT64",
+		Name:       "kubernetes.io/pod/volume/requested_bytes",
+	}
+
+	networkPodRxMD = &metricMetadata{
+		MetricKind: "CUMULATIVE",
+		ValueType:  "INT64",
+		Name:       "kubernetes.io/pod/network/bytes_rx",
+	}
+
+	networkPodTxMD = &metricMetadata{
+		MetricKind: "CUMULATIVE",
+		ValueType:  "INT64",
+		Name:       "kubernetes.io/pod/network/bytes_tx",
+	}
+
+	// Node metrics
+
+	cpuNodeCoreUsageTimeMD = &metricMetadata{
+		MetricKind: "CUMULATIVE",
+		ValueType:  "DOUBLE",
+		Name:       "kubernetes.io/node/cpu/core_usage_time",
+	}
+
+	cpuTotalCoresMD = &metricMetadata{
+		MetricKind: "GAUGE",
+		ValueType:  "DOUBLE",
+		Name:       "kubernetes.io/node/cpu/total_cores",
+	}
+
+	cpuAllocatableCoresMD = &metricMetadata{
+		MetricKind: "GAUGE",
+		ValueType:  "DOUBLE",
+		Name:       "kubernetes.io/node/cpu/allocatable_cores",
+	}
+
+	memoryNodeUsedBytesMD = &metricMetadata{
+		MetricKind: "GAUGE",
+		ValueType:  "INT64",
+		Name:       "kubernetes.io/node/memory/used_bytes",
+	}
+
+	memoryTotalBytesMD = &metricMetadata{
+		MetricKind: "GAUGE",
+		ValueType:  "INT64",
+		Name:       "kubernetes.io/node/memory/total_bytes",
+	}
+
+	memoryAllocatableBytesMD = &metricMetadata{
+		MetricKind: "GAUGE",
+		ValueType:  "INT64",
+		Name:       "kubernetes.io/node/memory/allocatable_bytes",
+	}
+
+	networkNodeRxMD = &metricMetadata{
+		MetricKind: "CUMULATIVE",
+		ValueType:  "INT64",
+		Name:       "kubernetes.io/node/network/bytes_rx",
+	}
+
+	networkNodeTxMD = &metricMetadata{
+		MetricKind: "CUMULATIVE",
+		ValueType:  "INT64",
+		Name:       "kubernetes.io/node/network/bytes_tx",
+	}
+
+	cpuNodeDaemonCoreUsageTimeMD = &metricMetadata{
+		MetricKind: "CUMULATIVE",
+		ValueType:  "DOUBLE",
+		Name:       "kubernetes.io/node_daemon/cpu/core_usage_time",
+	}
+
+	memoryNodeDaemonUsedBytesMD = &metricMetadata{
+		MetricKind: "GAUGE",
+		ValueType:  "INT64",
+		Name:       "kubernetes.io/node_daemon/memory/used_bytes",
+	}
+
+	// Old resource model metrics
+
+	uptimeLegacyMD = &metricMetadata{
```

Comment:
Adding legacy as a prefix, not in the middle of the name helps to distinguish and discourage better

Reply:
Done